library/ssl_tls.c

```diff
@@ -260,6 +260,68 @@ static int resize_buffer( unsigned char **buffer, size_t len_new, size_t *len_ol
 
     return 0;
 }
+
+static void handle_buffer_resizing( mbedtls_ssl_context *ssl, int downsizing,
+                                    uint32_t in_buf_new_len,
+                                    uint32_t out_buf_new_len )
```

Comment:
Why do `mbedtls_ssl_get_output_buflen` and `mbedtls_ssl_get_input_buflen` return `uint32_t`? The value is a buffer size. It is only used for comparisons with `size_t` values and to be passed to `resize_buffers` which wants a `size_t`. So I don't see any reason to use something other than `size_t`.

Reply:
Fixed.